starknet_py/proxy/contract_abi_resolver.py

```diff
@@ -85,26 +87,27 @@ async def resolve(self) -> AbiDictList:
             return await self.get_abi_for_address()
         return await self.resolve_abi()
 
-    async def get_abi_for_address(self) -> AbiDictList:
+    async def get_abi_for_address(self) -> Tuple[AbiDictList, int]:
         """
-        Returns abi of a contract directly from address.
+        Returns abi and cairo version of a contract directly from address.
 
         :raises ContractNotFoundError: when contract could not be found at address
         :raises AbiNotFoundError: when abi is not present in contract class at address
         """
         contract_class = await _get_class_at(address=self.address, client=self.client)
-        if isinstance(contract_class, SierraContractClass):
-            # TODO (#1012): Consider better handling
-            raise UnsupportedAbiError(
-                "Proxy resolver does not currently support Cairo1 ABIs."
-            )
+
         if contract_class.abi is None:
             raise AbiNotFoundError()
-        return contract_class.abi
 
-    async def resolve_abi(self) -> AbiDictList:
+        if isinstance(contract_class, SierraContractClass):
+            assert isinstance(contract_class.abi, str)
+            return json.loads(contract_class.abi), 1
+
+        return cast(AbiDictList, contract_class.abi), 0
```

Comment:
Why only cast to the type when the version is 0?

Reply:
Because json.loads already returns a `List[Dict]`